storehaus-kafka/src/main/scala/com/twitter/storehaus/kafka/KafkaStore.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2014 Twitter inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.twitter.storehaus.kafka
+
+import com.twitter.storehaus.WritableStore
+import com.twitter.util.{Time, Future}
+import java.util.Properties
+import org.apache.kafka.clients.producer.{ProducerRecord, KafkaProducer}
+import org.apache.kafka.common.serialization.Serializer
+
+/**
+  * Store capable of writing to a Kafka topic.
+  * @author Mansur Ashraf
+  * @since 11/22/13
+  */
+class KafkaStore[K, V](topic: String, props: Properties)
+  extends WritableStore[K, V] with Serializable {
+
+  private lazy val producer = new KafkaProducer[K, V](props)
+
+  /**
+    * Put a key/value pair in a Kafka topic
+    * @param kv (key, value)
+    * @return Future.unit
+    */
+  override def put(kv: (K, V)): Future[Unit] = Future {
+    val (key, value) = kv
+    producer.send(new ProducerRecord[K, V](topic, key, value))
```

Comment:
I think it is worth doing this right. This would mean that you would pass such a `JavaToScala` instance that has a thread inside into the class construction, and you could reuse them across `KafkaStore` instances or have `apply` methods to just allocate a new one (new thread) for each instance (and `release` it in the `close` method).